src/engine/Input/Pointer.ts

```diff
@@ -321,106 +327,254 @@ export class Pointers extends Class {
    }
 
    /**
-    * Propogates events to actor if necessary
+    * Propogates events thorugh event path if necessary
     */
-   public propogate(actor: Actor) {
-      var isNotUIActor = !Actors.isUIActor(actor);
-      var i: number = 0;
-      var len: number = 0;
-      var pointerEvent: PointerEvent;
-      var pointer: Pointer;
+   public propagate() {
+      this._propagatePointerEvent('pointerdown', this._pointerDown);
+      this._propagatePointerEvent('pointerup', this._pointerUp);
+      this._propagatePointerEvent('pointermove', this._pointerMove);
+      this._propagatePointerEvent('pointercancel', this._pointerCancel);
+      this._propagateWheelEvent('pointerwheel', this._wheel);
+   }
+
+   /**
+    * Verifies Actor pointer events accordingly to actor provided
+    * @param actor  Actor to be verified
+    */
+   public verifyPointerEvents(actor: Actor) {
+      this._verifyActorUnderPointers(actor, this._pointerDown);
+      this._verifyActorUnderPointers(actor, this._pointerUp);
+      this._verifyActorUnderPointers(actor, this._pointerMove);
+      this._verifyActorUnderPointers(actor, this._pointerCancel);
+      this._validateWheelEventPath(this._wheel, actor);
+   }
 
-      i = 0;
-      len = this._pointerDown.length;
+   private _verifyActorUnderPointers(actor: Actor, pointers: PointerEvent[]) {
```

Comment:
`verify` implies that the method is not side-effecting.

Let's refactor this method and break it into 2 methods:
* `_generateActorsUnderPointers()` That builds a list of pointer/actor groupings
* `_propagate(listOfPointerActor)` That propagates events through a given list